src/components/ListItem.jsx

```diff
@@ -1,5 +1,37 @@
 import './ListItem.css';
+import { updateItem } from '../api';
+import { sub } from 'date-fns';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listToken }) {
+	const handleUpdate = async (isChecked) => {
+		let day = Date.now();
+		let purchaseCounter = item.totalPurchases + 1;
+
+		try {
+			if (isChecked) {
+				const newData = { day, purchaseCounter };
+				await updateItem(listToken, item.id, newData);
```

Comment:
This is an async function, please wrap it in try/catch

Reply:
The logic that calls the updateItem is current wrapped inside a try/catch block

Reply:
You're right, I missed that 👍
I expected the try to wrap the `updateItem` directly, but this is okay too